app/src/marketplace/containers/EditProductPage2/useModal.jsx

```diff
@@ -0,0 +1,25 @@
+// @flow
+
+import { useContext, useMemo } from 'react'
+
+import { Context as ModalContext } from './Modal'
+
+export default function useModal(modalId: string) {
+    const { modals, openModal, closeModal } = useContext(ModalContext)
+
+    const isOpen = !!modals[modalId]
+
+    const value = isOpen && modals[modalId].value
+
+    const api = useMemo(() => ({
+        open: openModal.bind(null, modalId),
+        close: closeModal.bind(null, modalId),
+    }), [openModal, closeModal, modalId])
+
+    return useMemo(() => ({
+        modalId,
+        isOpen,
+        api,
+        value,
+    }), [value, modalId, isOpen, api])
+}
```

Comment:
Yet another modal API 🎉😸 We really need to standardise & centralise our modals. I like this API.

Reply:
Moved the modal to shared in https://github.com/streamr-dev/streamr-platform/pull/657/commits/f76f1590e5721ea116deaca7a91e717eac71425b and changed the editor to use that in https://github.com/streamr-dev/streamr-platform/pull/657/commits/4921590c240e6fcf7c1514b9ba8fb04fdf96eeee